### PR TITLE
Detection of deprecated constants

### LINKED ISSUE #58: Deprecated global constants are not discovered
See https://github.com/drupalcommerce/commerce/commit/e07e89819c21b472b7938637af003991b8fe6fad

### PR DESCRIPTION
**Pull request by mglaman**
Fixes #58

### CODE REVIEW

File: tests/src/DeprecationRulesTest.php [hunk 1]
```diff
@@ -19,19 +19,27 @@ public function testDeprecationRules(string $path, int $count, array $errorMessa
...
-                'Call to deprecated method url() of class Drupal.'
-            ]
-        ];
-        yield [
-            __DIR__ . '/../fixtures/drupal/core/lib/Drupal/Core/Entity/EntityManager.php',
-            1,
-            [
-                'Class Drupal\Core\Entity\EntityManager implements deprecated interface Drupal\Core\Entity\EntityManagerInterface.'
+                'Call to deprecated constant DATETIME_STORAGE_TIMEZONE.',
+                'Call to deprecated constant DATETIME_DATE_STORAGE_FORMAT.',
```
Comment (goba) [inline tests/src/DeprecationRulesTest.php]: These should be failing now as we should now get the deprecation message as well.

File: src/Rules/Deprecations/AccessDeprecatedConstant.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public function processNode(Node $node, Scope $scope): array
+    {
+        if (DeprecatedScopeHelper::isScopeDeprecated($scope)) {
+            return [];
+        }
+
+        // comment is not resolved.
+        $comment = $node->getDocComment();
```
Comment (goba) [inline src/Rules/Deprecations/AccessDeprecatedConstant.php]: Why do we even need the doc comment here? Is it to decide from that if the constant was deprecated?

File: src/Rules/Deprecations/AccessDeprecatedConstant.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        $constnatName = $this->broker->resolveConstantName($node->name, $scope);
```
Comment (goba) [inline src/Rules/Deprecations/AccessDeprecatedConstant.php]: Typo constnatName :)

File: src/Rules/Deprecations/AccessDeprecatedConstant.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    public function getNodeType(): string
+    {
+        return Node\Expr\ConstFetch::class;
+    }
+
+    public function processNode(Node $node, Scope $scope): array
+    {
+        if (DeprecatedScopeHelper::isScopeDeprecated($scope)) {
+            return [];
```
Comment (goba) [inline src/Rules/Deprecations/AccessDeprecatedConstant.php]: Is this to avoid reporting deprecated things in already deprecated scopes? Yay!

Comment (mglaman) [reply src/Rules/Deprecations/AccessDeprecatedConstant.php]: yep!

**Comment by mglaman** [comment]
Problems. nikic/php-parser doesn't seem to pick up comments on global constants.

Screen Shot 2019-07-29 at 5 17 04 PM

**Pull request merged**